### PR TITLE
transceivers-server: cleanup interface

### PR DESCRIPTION
**Pull request by Aaron-Hartwig**
Most of the interface here was built out to facilitate benchtop testing during initial development. Now that we have these interfaces proven out and connected via the UDP interface, we don't really need them as part of the general Idol server interface. I'm opting to remove them completely as otherwise it just doubles the surface area where we need to manage the module interactions. This will leave the UDP interface as the sole way to manage the transceivers going forward.

### CODE REVIEW

File: task/thermal/src/control.rs [hunk 1]
```diff
@@ -226,6 +226,7 @@ pub struct ThermalSensorErrors {
 }
 
 impl ThermalSensorErrors {
+    #[allow(dead_code)]
```
Comment (Aaron-Hartwig) [inline task/thermal/src/control.rs:229]: This is unrelated to the rest of this PR, it just cleans up a warning currently on `master`.

**Review by mkeeter — APPROVED** [review]
deleting code good
